public/lib/downloadHttpConcurrentProgress.js

```diff
@@ -77,11 +79,12 @@
      * @return error object
      */
     downloadHttpConcurrentProgress.prototype.onTestError = function (result) {
-        if (this._running) {
-            this.clientCallbackError(result);
-            clearInterval(this.interval);
-            this._running = false;
-        }
+      console.dir(result);
+      if (this._running) {
+         if ((Date.now() - this._beginTime) > this.testLength) {
```

Comment:
we might need a 'else' case here to send the error back if it is during the 12 sec